Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/utilities/AgoLaterUtil.java

```diff
@@ -0,0 +1,210 @@
+package com.microsoft.recognizers.text.datetime.utilities;
+
+import com.google.common.collect.ImmutableMap;
+import com.microsoft.recognizers.text.ExtractResult;
+import com.microsoft.recognizers.text.datetime.Constants;
+import com.microsoft.recognizers.text.datetime.extractors.IDateTimeExtractor;
+import com.microsoft.recognizers.text.datetime.parsers.DateTimeParseResult;
+import com.microsoft.recognizers.text.datetime.parsers.IDateTimeParser;
+import com.microsoft.recognizers.text.datetime.parsers.config.IDateParserConfiguration;
+import com.microsoft.recognizers.text.utilities.Match;
+import com.microsoft.recognizers.text.utilities.MatchGroup;
+import com.microsoft.recognizers.text.utilities.RegExpUtility;
+import com.microsoft.recognizers.text.utilities.StringUtility;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+public class AgoLaterUtil {
+    public static DateTimeResolutionResult parseDurationWithAgoAndLater(String text,
+                                                                        LocalDateTime referenceTime,
+                                                                        IDateTimeExtractor durationExtractor,
+                                                                        IDateTimeParser durationParser,
+                                                                        ImmutableMap<String, String> unitMap,
+                                                                        Pattern unitRegex,
+                                                                        IDateTimeUtilityConfiguration utilityConfiguration,
+                                                                        IDateParserConfiguration config) {
+        DateTimeResolutionResult ret = new DateTimeResolutionResult();
```

Comment:
Nitpick: Can we always add a blank line between method signature and implementation? It helps a lot in reviewing the code sometimes.